### ISSUE TITLE
demo1-jax.py

### ISSUE THREAD

**Issue by ismael-mendoza** [GalSim-developers/JAX-GalSim#92]
Showcase how to use jax functionality in jax galsim, specifically:

- [ ] jitting
- [ ] vmapping
- [ ] gradients

Potential cool extensions (potentially other demos)
- [ ] Derivatives with respect to shear `d Image / d Shear` 
- [ ] running gradient based samplers with jax-galsim (eg blackjax, numpyro)
- [ ] jax metacal (see unit test)
- [ ] full stamp image drawing with jax galsim (see unit test)
